### PR TITLE
Add default response timing middleware

### LINKED ISSUE #4: Add response timing middleware
The first thing I did in my app with this client was to add `before_request` and `after_request` handlers to get response time metrics into datadog. Seems like it might be a decent/common functionality people would want?

The gist of it looks like this:

``` python
import time

from flask import Flask, g, request
from flask.ext.datadog import StatsD

app = Flask(__name__)
statsd = StatsD(app=app)

def before_request():
    g.request_start_time = time.time()

def after_request(response):
    # Return early if we don't have the start time
    if hasattr(g, 'request_start_time'):
        return response

    # Get elapsed time in milliseconds
    elapsed = time.time() - g.request_start_time
    elapsed = int(round(1000 * elapsed))

    # Collect request/response tags
    tags = [
        'endpoint:{endpoint}'.format(endpoint=request.endpoint),
        'request_method:{method}'.format(method=request.method.lower()),
        'status_code:{status_code}'.format(status_code=response.status_code),
    ]

    # Record our response time metric
    statsd.timing('flask.response.time', elapsed, tags=g.request_tags + tags)

    # Return the original unmodified response
    return response

app.before_request(before_request)
app.after_request(after_request)
```

This might be something we can do by default when running `init_app`, or maybe just a method `setup_middleware(app)` for handling it.

Does this seem like a reasonable thing to try and tackle, or should we leave this middleware stuff as an exercise for the implementer?

### PR DESCRIPTION
**Pull request by brettlangdon**
Fixes #4 

This PR adds default response timing middleware to `Flask-Datadog`, which will record a metric `flask.response.time` after every request. As well, it provides a means for managing per-request tags.

Example:

``` python
from flask import Flask
from flask.ext.datadog import StatsD

app = Flask(__name__)
statsd = StatsD(app=app)


@app.route('/')
def root():
    # Add a new tag `thanks` only to this request
    statsd.add_request_tags(['thanks'])

    # Send a new metric and use the available request tags that have been set
    statsd.increment('endpoint.metric', tags=statsd.get_request_tags())

    return 'thanks'


app.debug = True
app.run()
```

### CODE REVIEW

File: flask_datadog.py [hunk 1]
```diff
@@ -98,6 +111,85 @@ def decr(self, *args, **kwargs):
...
+        """
+        Flask-Datadog middleware handle for before each request
+        """
+        # Set the request start time
+        g.flask_datadog_start_time = time.time()
+        g.flask_datadog_request_tags = []
+
+        # Add some default request tags
+        if self.config['DATADOG_RESPONSE_AUTO_TAG']:
+            self.add_request_tags(['endpoint:%s' % (str(request.endpoint).lower(), ),
```
Comment (volker48) [inline flask_datadog.py]: Could you add a config option to control the tag name for the endpoint? We have a bunch of code that is using `endpoint_name:` and I don't want to break a bunch of monitors and dashboards.

Comment (brettlangdon) [inline flask_datadog.py]: Sure, not a problem, assume we should do the same for `method` as well, if we let one be configurable, might as well make the other as well?

/me knows the pain of trying to reconfigure a bunch of dashboards

**Comment by volker48** [comment]
> Sure, not a problem, assume we should do the same for method as well, if we let one be configurable, might as well make the other as well?

Yeah good thinking.

**Comment by brettlangdon** [comment]
I used the code here to allow configuring the tag names for the added tags as well as using `format` over `%`.

**Pull request merged**